src/character.cpp

```diff
@@ -596,6 +596,54 @@ std::vector<item_location> Character::nearby( const std::function<bool(const ite
     return res;
 }
 
+long int Character::i_add_to_container(const item &it)
+{
+    long int charges = it.charges;
+    
+    if( it.is_ammo() && (!it.has_flag( "TEMP_UNLOADING" ) || (it.has_flag( "TEMP_UNLOADING" ) && charges == 1) ) ) {
+        const itype_id item_type = it.typeId();
+        
+        auto add_to_container = [it](item &container) {
```

Comment:
You sure this capture won't create a copy of the item instead of properly referencing it?